test/cheap_ruler.cpp

```diff
@@ -13,7 +13,10 @@ class CheapRulerTest : public ::testing::Test {
 };
 
 void assertErr(double expected, double actual, double maxError) {
-    double error = std::abs((actual - expected) / expected);
+    // Add a negligible fraction to make sure we
+    // don't divide by zero.
+    double error = std::abs((actual - expected) /
+            (expected == 0. ? expected + 0.000001 : expected));
```

Comment:
Since this is only a hypothetical situation, I'd just do `(expected == 0. ? 1. : expected));`

Reply:
I tried first failing the test in case of `expected` being zero, but I found out we do have zero as expected result.

I'm not sure how it was not crashing. I suppose because of fp rounding errors.

Reply:
I mean not failing, just simplifying the expression so that it effectively becomes `error = std::abs(actual)` if `expected` is `0`.